### PR TITLE
server: reorganize server start up.

### PR DESCRIPTION
**Pull request by disksing**
Change 3 phase startup process `create/new, runEtcd, runServer` to 2 phase `create, run`.

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @huachaohuang

File: server/server.go [hunk 1]
```diff
@@ -44,103 +44,76 @@ const (
...
-
-	go systimemon.StartMonitor(time.Now, func() {
-		log.Errorf("system time jumps backward")
-		timeJumpBackCounter.Inc()
-	})
-	return s, nil
+	// For tso, set after pd becomes leader.
+	ts            atomic.Value
+	lastSavedTime time.Time
+  // For resign notify.
```
Comment (huachaohuang) [inline server/server.go]: Is this format right?

File: server/server.go [hunk 2]
```diff
@@ -44,103 +44,76 @@ const (
...
 
 	s := &Server{
-		cfg:           cfg,
-		scheduleOpt:   newScheduleOption(cfg),
-		isLeaderValue: 0,
-		closed:        1,
-		resignCh:      make(chan struct{}),
+		cfg:         cfg,
+		scheduleOpt: newScheduleOption(cfg),
+    resignCh:      make(chan struct{}),
```
Comment (huachaohuang) [inline server/server.go]: And this?

**Comment by disksing** [comment]
CI fails due to data race. I'll restart it and fix race in another PR.

**Pull request merged**